### ISSUE TITLE
Interstitial screen release review issues

### ISSUE THREAD

**Issue by amal-k-joy** [carbon-design-system/ibm-products#7699]
Fix the issues from[ Interstitial screen release review](https://github.com/carbon-design-system/ibm-products/issues/7607)

- 2 a11y violations in accessibility tab needs to be fixed
- interstitialAriaLabel naming convention to be more generic ?? agree naming to ariaLabel
- aligning isOpen -> open to match tearsheet ??
- Add devtools attribute to sub components (header, body, footer)
- Pass rest props in both footer and header sub component
- Export types for sub components from `InterstitialScreen/index.ts`
- Ensure all types from base IntersitialScreen and it's sub components are exported
- Revisit if it is absolutely necessary to use `!important`s in `_interstitial-screen.scss`
- A block class variable is defined in the based and all sub-components, to clean it up a little bit you could just export it from the based and not have to re-define it 
- The button appears to be overflowing its container by 1 px. in some variations.

<img width="873" alt="Image" src="https://github.com/user-attachments/assets/1ee8b557-692d-4ffa-84ac-5d086df38c23" />
